### PR TITLE
fix: support top multi compiler options

### LINKED ISSUE #2873: parallelism not work
# Bug report

**What is the current behavior?**

https://webpack.docschina.org/configuration/configuration-types/#parallelism

parallelism does not work

**If the current behavior is a bug, please provide the steps to reproduce.**

index.js

```js
console.log(`test ${Math.random()}`)
console.log(`test ${Math.random()}`)
console.log(`test ${Math.random()}`)
console.log(`test ${Math.random()}`)
console.log(`test ${Math.random()}`)
console.log(`test ${Math.random()}`)
```

webpack.config.js

```js
const path = require("path")
const SimpleProgressWebpackPlugin = require("simple-progress-webpack-plugin")

const configs = []

for (let i = 0; i < 3; i++) {
  configs.push({
    name: `build${i}`,
    entry: "./index",
    output: {
      path: path.join(__dirname, "dist"),
      filename: `build${i}.js`,
    },
    plugins: [
      new SimpleProgressWebpackPlugin({
        name: `build${i}`,
        format: "simple",
      }),
    ],
  })
}

module.exports = configs
module.exports.parallelism = 1

```

output:

```shell
build0

  ❯ Compile modules
build1

  ❯ Compile modules
build2

  ❯ Compile modules
  ❯ Build modules
  ❯ Build modules
  ❯ Build modules
  ❯ Optimize modules
  ❯ Optimize modules
  ❯ Optimize modules
  ❯ Emit files

Finished after 0.244 seconds.

  ❯ Emit files

Finished after 0.242 seconds.

  ❯ Emit files

Finished after 0.254 seconds.

build0
  ❯ Emit files

Finished after 0.000 seconds.

build1
  ❯ Emit files

Finished after 0.000 seconds.

build2
  ❯ Emit files

Finished after 0.001 seconds.

build0:
  asset build0.js 222 bytes [compared for emit] [minimized] (name: main)
  ./index.js 222 bytes [built] [code generated]

  WARNING in configuration
  The 'mode' option has not been set, webpack will fallback to 'production' for this value.
  Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
  You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

  build0 (webpack 5.47.1) compiled with 1 warning in 237 ms

build1:
  asset build1.js 222 bytes [compared for emit] [minimized] (name: main)
  ./index.js 222 bytes [built] [code generated]

  WARNING in configuration
  The 'mode' option has not been set, webpack will fallback to 'production' for this value.
  Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
  You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

  build1 (webpack 5.47.1) compiled with 1 warning in 230 ms

build2:
  asset build2.js 222 bytes [compared for emit] [minimized] (name: main)
  ./index.js 222 bytes [built] [code generated]

  WARNING in configuration
  The 'mode' option has not been set, webpack will fallback to 'production' for this value.
  Set 'mode' option to 'development' or 'production' to enable defaults for each environment.
  You can also set it to 'none' to disable any default behavior. Learn more: https://webpack.js.org/configuration/mode/

  build2 (webpack 5.47.1) compiled with 1 warning in 215 ms
```

**What is the expected behavior?**

**Other relevant information:**
webpack version:  5.47.1
Node.js version: v14.16.1
Operating System: ubuntu
Additional tools: "webpack-cli": "^4.7.2"

### PR DESCRIPTION
**Pull request by alexander-akait**
**What kind of change does this PR introduce?**

bugfix

**Did you add tests for your changes?**

yes

**If relevant, did you update the documentation?**

No

**Summary**

fixes #2873

**Does this PR introduce a breaking change?**

No

**Other information**

We should refactor our logic for multiple `--config`, ideally I think we should run `merge` when multiple configurations provided, but it is breaking change and should be solved for the next major release

### CODE REVIEW

File: packages/webpack-cli/lib/webpack-cli.js [hunk 1]
```diff
@@ -1521,107 +1521,124 @@ class WebpackCLI {
...
+    }
+
+    async resolveConfig(options) {
         const config = { options: {}, path: new WeakMap() };
 
         if (options.config && options.config.length > 0) {
-            const evaluatedConfigs = await Promise.all(
-                options.config.map(async (value) =>
-                    evaluateConfig(await loadConfig(path.resolve(value)), options.argv || {}),
+            const loadedConfig = await Promise.all(
```
Comment (snitin315) [inline packages/webpack-cli/lib/webpack-cli.js:1608]:
```suggestion
            const loadedConfigs = await Promise.all(
```

File: packages/webpack-cli/lib/webpack-cli.js [hunk 2]
```diff
@@ -1521,107 +1521,124 @@ class WebpackCLI {
...
 
-            evaluatedConfigs.forEach((evaluatedConfig) => {
-                if (Array.isArray(evaluatedConfig.options)) {
-                    evaluatedConfig.options.forEach((options) => {
-                        config.options.push(options);
-                        config.path.set(options, evaluatedConfig.path);
+            loadedConfig.forEach((loadedConfig) => {
+                const isArray = Array.isArray(loadedConfig.options);
+
+                // TODO we should run webpack multiple times when the `--config` options has multiple values without `--merge`, need to solve for the next major release
```
Comment (snitin315) [inline packages/webpack-cli/lib/webpack-cli.js:1619]:
```suggestion
                // TODO we should run webpack multiple times when the `--config` options have multiple values with `--merge`, need to solve for the next major release
```

File: packages/webpack-cli/lib/webpack-cli.js [hunk 3]
```diff
@@ -1521,107 +1521,124 @@ class WebpackCLI {
...
             );
 
             config.options = [];
 
-            evaluatedConfigs.forEach((evaluatedConfig) => {
-                if (Array.isArray(evaluatedConfig.options)) {
-                    evaluatedConfig.options.forEach((options) => {
-                        config.options.push(options);
-                        config.path.set(options, evaluatedConfig.path);
+            loadedConfig.forEach((loadedConfig) => {
```
Comment (snitin315) [inline packages/webpack-cli/lib/webpack-cli.js:1616]:
```suggestion
            loadedConfigs.forEach((loadedConfig) => {
```

**Review by snitin315 — APPROVED** [review]
LGTM

File: packages/webpack-cli/lib/webpack-cli.js [hunk 4]
```diff
@@ -1521,107 +1521,124 @@ class WebpackCLI {
...
-        const loadConfig = async (configPath) => {
-            const { interpret } = this.utils;
-            const ext = path.extname(configPath);
-            const interpreted = Object.keys(interpret.jsVariants).find(
-                (variant) => variant === ext,
-            );
+    async loadConfig(configPath, argv = {}) {
+        const { interpret } = this.utils;
+        const ext = path.extname(configPath);
+        const interpreted = Object.keys(interpret.jsVariants).find((variant) => variant === ext);
```
Comment (anshumanv) [inline packages/webpack-cli/lib/webpack-cli.js:1527]: can just use `interpret.jsVariants[ext]` ?

Comment (anshumanv) [reply packages/webpack-cli/lib/webpack-cli.js:1527]: oh it is existing logic, maybe we can improve it late

**Review by anshumanv — APPROVED** [review]

**Pull request merged**